src/hooks/tokenOwner/escrowContractUsers.ts

```diff
@@ -0,0 +1,46 @@
+import { createOwner, TitleEscrowOwner } from "@govtechsg/oa-token";
+
+import { getLogger } from "../../utils/logger";
+
+import { useCallback, useEffect, useState } from "react";
+import { useEthereumTransactionState, TransactionStateStatus, StateType } from "../helpers";
+import { useProvider } from "../provider";
+const { trace, error } = getLogger("hooks:escrowContractUsers");
+
+type UseEscrowContractUsers = { state: StateType; holderAddress: string; beneficiaryAddress: string };
+
+export const useEscrowContractUsers = ({
+  escrowContractAddress
+}: {
+  escrowContractAddress: string;
+}): UseEscrowContractUsers => {
+  trace(`document to initialize ${JSON.stringify(document)}`);
+  const [holderAddress, setHolderAddress] = useState("");
+  const [beneficiaryAddress, setBeneficiaryAddress] = useState("");
+  const { web3Provider } = useProvider();
+  const [state, dispatch] = useEthereumTransactionState();
+  const setDispatchCallback = useCallback(dispatch, []);
+
+  const getApprovedEscrowContractUsers = useCallback(async () => {
+    const titleEscrowInstance = await createOwner({ address: escrowContractAddress, web3Provider });
+    if (!(titleEscrowInstance instanceof TitleEscrowOwner)) throw new Error("Address is not an Escrow contract");
+    const [beneficiary, holder] = await Promise.all([titleEscrowInstance.beneficiary(), titleEscrowInstance.holder()]);
+    trace(`approved beneficiary: ${JSON.stringify(beneficiary)}, approved holder: ${holder}`);
+    setHolderAddress(holder);
+    setBeneficiaryAddress(beneficiary);
+  }, [escrowContractAddress, web3Provider]);
+
+  useEffect(() => {
+    (async () => {
```

Comment:
let's avoid iife, it doesnt help for readability with all the noisy syntax

Reply:
fixed